test/RouteCollectorTest.php

```diff
@@ -185,13 +185,81 @@ public function testCreatingHttpRouteMethodWithExistingPathButDifferentMethodCre
         $this->assertSame($middleware, $test->getMiddleware());
     }
 
+    public function testCreatingHttpRouteWithExistingPathShouldBeLinear()
+    {
+        $start = microtime(true);
+        foreach (range(1, 10) as $item) {
+            $this->collector->get("/bar$item", $this->noopMiddleware);
+        }
+        $baseDuration = microtime(true) - $start;
+        $this->collector = new RouteCollector($this->router->reveal());
+
+        $start = microtime(true);
+        foreach (range(1, 10000) as $item) {
+            $this->collector->get("/foo$item", $this->noopMiddleware);
+        }
+
+        $duration = microtime(true) - $start;
+        $expectedDuration = $baseDuration * 1000;
+        $error = 30 * $expectedDuration / 100;
+        $this->assertTrue(
+            $expectedDuration + $error > $duration,
```

Comment:
Why not use `assertGreaterThan($expectedDuration + $error, $duration, ...)` here?

Also, more to the point: shouldn't this comparison be... the opposite? The expected side is essentially 300x the base duration; if it comes in less, why is that a problem? Don't we want to check that the actual duration is less than or equal to that value? (Shorter duration is better here...)

(Also, where did you get that expected interval calculation from?)

Reply:
We measure base time for 10 route additions and expecting that execution time is linear `base_time*x = actual_time`. But in real world we have interrupts and concurrency, if on my machine this equation is really very very close to `$expectedDuration ≈ $duration` on machine with high concurrency this is not as stable (ci environment), so i decided to add error, its size is 30% this is empirical for ci tests pass. So `$duration` should be less or equal than `$expectedDuration + $error`. This is still true that this is linear because lireraly we a saying that `$duration = 1.3*n*$baseDuration` before that PR it was `$duration = n^2 - n/2`.
As of `assertGreaterThan` i can use it if you insist, but i think current variant is more readable in sense of equation that need to be true.